File: Leetcode/Easy/PYTHON/26. Remove Duplicates from Sorted Array.py
from typing import List
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        pos=0
        for i in range(1,len(nums)):
            if(nums[i]!=nums[pos]):
                nums[pos+1]=nums[i]
                pos=pos+1
        return pos+1

#Short Thought Process:
# We maintain a pointer 'pos' to track the position of the last unique element.
# We iterate through the array starting from the second element.
# Whenever we find a new unique element (i.e., an element different from nums[pos]),
# we place it at the next position (pos + 1) and update 'pos'.
# Finally, we return pos + 1, which represents the count of unique elements in the
# modified array.

# Problem Link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/

File: Leetcode/Easy/CPP/26. Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size()==0) return 0;
        int pos=0;
        for(int i=1;i<nums.size();i++){
            if(nums[i]!=nums[pos]) swap(nums[++pos],nums[i]);
        }
        return pos+1;
    }
};

/* Short Thought Process: Since the array is sorted, duplicates will be adjacent. We maintain a pointer 'pos' to track the position of the last unique element found. As we iterate through the array, whenever we find a new unique element, we increment 'pos' and swap the current element with the element at 'pos'. Finally, we return 'pos + 1' as the count of unique elements. */